miner/src/pool/scoring.rs

```diff
@@ -92,33 +92,69 @@ impl<P> txpool::Scoring<P> for NonceAndGasPrice where P: ScoredTransaction + txp
 	}
 
 	fn update_scores(&self, txs: &[txpool::Transaction<P>], scores: &mut [U256], change: scoring::Change) {
-		use self::scoring::Change;
-
-		match change {
-			Change::Culled(_) => {},
-			Change::RemovedAt(_) => {}
-			Change::InsertedAt(i) | Change::ReplacedAt(i) => {
-				assert!(i < txs.len());
-				assert!(i < scores.len());
-
-				scores[i] = *txs[i].transaction.gas_price();
-				let boost = match txs[i].priority() {
-					super::Priority::Local => 15,
-					super::Priority::Retracted => 10,
-					super::Priority::Regular => 0,
-				};
-				scores[i] = scores[i] << boost;
+
+		let boost = |i: usize| {
+			match txs[i].priority() {
+				super::Priority::Local => 15,
+				super::Priority::Retracted => 10,
+				super::Priority::Regular => 0,
+			}
+		};
+
+		let event = |scores: &mut [U256]| {
+			for (score, tx) in scores.iter_mut().zip(txs) {
+				if !tx.priority().is_local() {
+					*score = *score >> 3;
+				}
+			}
+		};
+
+		match self.0 {
+			PrioritizationStrategy::GasPriceOnly => {
+				match change {
+					Change::Culled(_) => {},
+					Change::RemovedAt(_) => {}
+					Change::InsertedAt(i) | Change::ReplacedAt(i) => {
+						// calculate base score + boost
+						assert!(i < txs.len());
+						assert!(i < scores.len());
+						scores[i] = *txs[i].transaction.gas_price() << boost(i);
+					},
+					// We are only sending an event in case of penalization.
+					// So just lower the priority of all non-local transactions.
+					Change::Event(_) => event(scores),
+				}
 			},
-			// We are only sending an event in case of penalization.
-			// So just lower the priority of all non-local transactions.
-			Change::Event(_) => {
-				for (score, tx) in scores.iter_mut().zip(txs) {
-					// Never penalize local transactions.
-					if !tx.priority().is_local() {
-						*score = *score >> 3;
+			PrioritizationStrategy::Consecutive => {
+				let is_consecutive = |i: usize| {
+					if (txs[i].nonce() + U256::one()) == txs[i+1].nonce() {
+						U256::one()
+					} else {
+						U256::zero()
 					}
+				};
+				match change {
+					Change::Culled(_) => {},
+					Change::RemovedAt(_) => {}
```

Comment:
So actually it's fine to ignore those events in `GasPriceOnly`, but with `Consecutive` strategy we should recalculate as well.